sdk/core/Azure.Core/tests/TelemetryDetailsTests.cs

```diff
@@ -115,19 +108,23 @@ public void AsciiDoesNotEncode(string input, string output)
                 version = version.Substring(0, hashSeparator);
             }
 
-            var target = new TelemetryDetails(typeof(TelemetryDetailsTests).Assembly, default, mockRuntimeInformation);
+            // Verify that the assembly name was transformed to Azure SDK format
+            string userAgent = target.ToString();
```

Comment:
We should assert on the full string. No need to have separate assertions for each piece.

Reply:
We should assert on the full string. No need to have separate assertions for each piece.